### PR TITLE
Added all desired methods and tests

### PR DESCRIPTION
**Pull request by mra48**
For anyone using these methods here are some things to keep in mind.
1. I Changed type from enum to String because enum was too difficul to
   deal with and I wouldnt have gotten much done if I tried to implement
   it. So now the types are "Annual", "Perennial", "Tree" as Strings.
2. The first method you should call before others are stringToGarden
   (unless you already have a garden from a saved file, etc...) because
   this will help avoid null pointer exceptions.
3. Also, if you call addSpecies(String speciesName), then make sure to
   set all of the other parameters first before calling gardenToString
   because this will also help avoid null pointer exceptions
4. The String format of a garden is listed above the first method in
   garden.java but ill copy it here for reference

format for the string representation of the garden->
# species-speciesName-speciesDescription-speciesType (ex: 0 for annual, 1

for perennial, 2 for tree)-speciesPlantDate(using this format:MM/dd/yyyy
HH:mm:ss)-speciesPruneDate(using this format:MM/dd/yyyy
HH:mm:ss)-speciesSunLevel-speciesColor-speciesSize-data for next
species-#plants-plantx-planty-plantSpecies-data for next plant

### CODE REVIEW

**Comment by chuckie512** [comment]
I'll start taking a look at it right now.

Also: for future reference it's considered good practice to have more than 1 commit for a diff this large

**Comment by chuckie512** [comment]
I'm seeing build errors right away, looks like you didn't keep old methods in for capability

**Comment by chuckie512** [comment]
the @ Test annotation you used doesn't work in the android environment.  SmallTest MediumTest or LargeTest should be used instead (based upon expected runtime of the test)

**Comment by chuckie512** [comment]
minor note: your test file has unused variable warnings, those should be quick to cleanup

**Comment by libertylocked** [comment]
I don't think we need to keep the old deprecated methods. Update the code in garden app.
Because eventually we need to get rid of the deprecated methods - submitting another PR takes more time

**Comment by chuckie512** [comment]
since I don't have any access to the fork repo, here's my changes (that only get it to compile) as a .patch that @mra48 will have to add himself

https://drive.google.com/file/d/0B-nwIqB7rekDdWhYV0pTdGNoRGM/view?usp=sharing

File: GardenApp/app/src/main/java/com/cs1530_group1/gardenapp/Garden.java [hunk 1]
```diff
@@ -58,178 +66,642 @@ public String GardenToString()
...
+    	
+    	if (s != null)
+    	{
+        	return s.size;
+    	}
+    	
+        return 0;
+    }
+    
+    /* Sprint 2 methods */
```
Comment (chuckie512) [inline GardenApp/app/src/main/java/com/cs1530_group1/gardenapp/Garden.java:483]: keeping all this commented out, unused code in our codebase is pretty ugly, I suggest removing it before it gets merged into the master 

lines 483+ just seem unneeded

Comment (libertylocked) [inline GardenApp/app/src/main/java/com/cs1530_group1/gardenapp/Garden.java:483]: Yeah one can easily use ~~myGarden.getSpeciesNames().length~~  myGarden.getSpecies("myspecies").size

Comment (chuckie512) [inline GardenApp/app/src/main/java/com/cs1530_group1/gardenapp/Garden.java:483]: oops not what I meant @LibertyLocked , I mean lines 483+ are just commented out code

**Comment by chuckie512** [comment]
from the testing side of things @mra48 you don't seem to ever start from an empty garden, that seems like an important test case

File: GardenApp/app/src/main/java/com/cs1530_group1/gardenapp/Garden.java [hunk 2]
```diff
@@ -58,178 +66,642 @@ public String GardenToString()
...
-    }    
-	
-	/**
-	 * Returns a list of current species
-	 * <p>
-	 * 
-	 * @return String array of species
-	 */		
-    public String[] ListSpecies()
+        return newSpecies;
```
Comment (chuckie512) [inline GardenApp/app/src/main/java/com/cs1530_group1/gardenapp/Garden.java:201]: Lines: 188-201

This doesn't seem like an appropriate name for what the method returns

EDIT: these line comments seem to be better in theory than practice -_-

**Comment by chuckie512** [comment]
@grselker copied the fork to a branch so we can make changes to the code before merging.

see branch: feature/UpdatedGardenInterface

**Pull request closed without merging**